MapboxVision/Helpers/DeviceChecker.swift

```diff
@@ -7,32 +7,36 @@
 //
 
 import Foundation
-import UIKit
 import MapboxVisionNative
+import UIKit
 
-private let iPhoneName = "iPhone"
-private let iPhoneMinModel = 10 // meaning iPhone 8/8Plus/X
+enum DeviceModel {
+    enum Name {
+        static let iPhone = "iPhone"
+    }
+
+    enum MajorNumber {
+        static let minIphoneVersionWithHighPerformance = 10 // "10" corresponds to 8/8 Plus/X.
+        static let maxIphoneVersionWithHighPerformance = 11 // "11" corresponds to XR/XS/Xs Max.
+    }
+}
 
 extension UIDevice {
-    var isTopDevice: Bool {
-        var prefix: String = ""
-        var minModel: Int = 0
-        
-        var modelID = self.modelID
-        
-        if modelID.hasPrefix(iPhoneName) {
-            prefix = iPhoneName
-            minModel = iPhoneMinModel
-        }
-        
-        guard !prefix.isEmpty, minModel > 0 else { return false }
-        
-        modelID.removeFirst(prefix.count)
-        
-        if let majorVersion = modelID.split(separator: ",").first, let majorNumber = Int(majorVersion) {
-            return majorNumber == minModel
+     // By default we consider next-gen iPhone as a uncapable to run models with high performance.
+     // This is due to potential issues with temperature during operational mode.
+     // We must do proper testing before upgrading `maxIphoneWithHighPerformance` version.
+    var isHighPerformanceIphone: Bool {
```

Comment:
Consider `isHighPerformance` as a more generic name.

Reply:
I thought about that. For now I'd propose to leave `isHighPerformanceIphone` as it reflects the actual behavior. Client's code at the moment of development can't figure out if we support `iPad`. In case we decide to add logic for `iPad` we definitely should rename it. What do you think? Does that make sense?

Reply:
At the moment client code doesn't take into consideration what type of the device is used. It's only important whether it is high performance or not.
From the top of my head there's two ways of how this may be developed when we add iPad support:
- we separate iPads into high and low performance and use the same values for FPS. In this case method `isHighPerformance` still makes sense
- iPads will have different matrix of performance in terms of FPS values. In this case the method is useless regardless of its name. Instead we should get smth like tuple of device type and its performance bracket.

Reply:
Agree, make sense.